File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/search_objects/volunteer_activity_search_object.dart
class VolunteerActivitySearchObject {
  String? title;
  int? locationId;
  int? status;

  String? sortBy;

  int? page;
  int? pageSize;

  VolunteerActivitySearchObject({
    this.title,
    this.locationId,
    this.status,
    this.sortBy,
    this.page,
    this.pageSize,
  });

  Map<String, dynamic> toJson() {
    return {
      "title": title,
      "locationId": locationId,
      "status": status,
      "sortBy": sortBy,
      "page": page,
      "pageSize": pageSize,
      "includeTotalCount": true,
    };
  }
}

File: eAnimalShelter/eAnimalShelter.Model/SearchObjects/VolunteerActivitySearchObject.cs
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Model.SearchObjects
{
    public class VolunteerActivitySearchObject : BaseSearchObject
    {
        public string? Title { get; set; }

        public int? LocationId { get; set; }

        public ActivityStatus? Status { get; set; }

        public DateTime? StartDateTimeFrom { get; set; }

        public DateTime? StartDateTimeTo { get; set; }
    }
}
